code/__HELPERS/matrices/transform_matrix.dm

```diff
@@ -84,3 +84,47 @@
  */
 /proc/transform_matrix_construct(a = 1, b, c, d = 1, e, f)
 	return matrix(a, b, c, d, e, f)
+
+/atom/proc/shake_animation(var/intensity = 8) //Makes the object visibly shake
+	var/initial_transform = new/matrix(transform)
+	var/init_px = pixel_x
+	var/shake_dir = pick(-1, 1)
+	var/rotation = 2+soft_cap(intensity, 1, 1, 0.94)
+	var/offset = 1+soft_cap(intensity*0.3, 1, 1, 0.8)
+	var/time = 2+soft_cap(intensity*0.3, 2, 1, 0.92)
+	animate(src, transform=turn(transform, rotation*shake_dir), pixel_x=init_px + offset*shake_dir, time=1)
+	animate(transform=initial_transform, pixel_x=init_px, time=time, easing=ELASTIC_EASING)
+
+/*
+	This proc makes the input taper off above cap. But there's no absolute cutoff.
+	Chunks of the input value above cap, are reduced more and more with each successive one and added to the output
+	A higher input value always makes a higher output value. but the rate of growth slows
+*/
+/proc/soft_cap(var/input, var/cap = 0, var/groupsize = 1, var/groupmult = 0.9)
```

Comment:
why is this in this file? whatever

Reply:
That is probably a mistake on my part. Lemme fix that.

Reply:
Moved to a separate file which is how the other ports had it done.